library/core/src/fmt/num.rs

```diff
@@ -336,20 +421,25 @@ macro_rules! impl_Display {
             unsafe {
                 loop {
                     curr -= 1;
-                    buf_ptr.add(curr).write((n % 10) as u8 + b'0');
+                    buf[curr].write((n % 10) as u8 + b'0');
                     n /= 10;
 
                     if n == 0 {
                         break;
                     }
                 }
             }
+            cur
```

Comment:
Should this be `curr`? How does this even compile?

Reply:
Oh, that's because of `optimize_for_size`. I'm surprised it's not tested in CI.